### PR TITLE
Fixes #15055: Add support for old OS in at least ncf_tests

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/15055

### CODE REVIEW

**Comment by Fdall** [comment]
Commit modified

File: scenario/ncf_tests.py [hunk 1]
```diff
@@ -13,6 +13,18 @@
 
 # Test begins, register start time
 start(__doc__)
+need_external_setup = ["ubuntu12_04", "ubuntu12_10", "ubuntu13_04"]
+
+def execute(target, command):
+  shell("echo " + command, live_output=True)
```
Comment (VinceMacBuche) [inline scenario/ncf_tests.py]: Do you want to keep the echo here ?

Comment (Fdall) [reply scenario/ncf_tests.py]: Yes

File: scenario/ncf_tests.py [hunk 2]
```diff
@@ -13,6 +13,18 @@
...
+  shell_on(target, command, live_output=True)
+
+def external_setup():
+  tag = get_param("tag", "")
+  vagrant_shared = scenario.platform.hosts['agent'].info['sync_file'] if 'sync_file' in scenario.platform.hosts['agent'].info else '/vagrant'
+  execute("localhost", "$(pwd)/scripts/external_setup_host " + scenario.pf + " " + tag)
+  execute("agent", "mv " + vagrant_shared + "/external_files/" + scenario.pf + " /tmp/")
```
Comment (Fdall) [inline scenario/ncf_tests.py]: missing /

File: scripts/external_setup_host [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+PLATFORM_NAME=$1
+TAG=$2
+
+DIR=$(dirname "${BASH_SOURCE[0]}")
+DST=${DIR}/../external_files/${PLATFORM_NAME}
+mkdir -p ${DST}
+
+# Get correct ncf branch
+git clone -b branches/rudder/${TAG} --single-branch https://github.com/Normation/ncf.git ${DST}/ncf
+wget -O ${DST}/rudder-setup https://repository.rudder.io/tools/ncf-setup
```
Comment (Fdall) [inline scripts/external_setup_host]: ${DST}/ncf-setup

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
OK, merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**